### ISSUE TITLE
Question on notation for Equation 5.6

### ISSUE THREAD

**Issue by Patrick-Wen** [dvgodoy/PyTorchStepByStep#47]
Here is Eq 5.6 in the book:
![Snipaste_2024-03-19_15-53-50](https://github.com/dvgodoy/PyTorchStepByStep/assets/100295693/fd4130cd-ef8f-483e-9287-c1de9e8d9ecf)

It is stated that "C stands for the number of classes". I think the N<sub>c</sub>, which represents the number of cases in the c<sup>th</sup> class, should be replaced by C. N<sub>c</sub> is simply irrelevant here since softmax is calucated per individual.

Please correct me if I am wrong. Thank you.

Patrick Wen

**Comment by dvgodoy** [comment]
Hi @Patrick-Wen,

Thank you for reporting this. 

You're absolutely right, it should be "C-1" instead of "Nc-1" in the upper part of the summation. I will fix it in the notebook and add to the list of typos to be addressed in a future revision.

Thanks for supporting my work :-)

Best,
Daniel

**Comment by Patrick-Wen** [comment]
As my reading of the book continues, I may wish to bring up some possible typos I figured out along the way.

1. In Page 530 of the book establishing the class Inception, here I suspect 2@HxW might have been mistakenly put as 1@HxW:
![Snipaste_2024-03-27_09-51-15](https://github.com/dvgodoy/PyTorchStepByStep/assets/100295693/03f872ae-5a8d-441d-9c2d-f6bb2708f195)

2. Page 695-6 put: "The hidden state, however, was entirely handled by the model itself using its hidden_state attribute", where hidden_state was printed in a different font from other texts, suggesting hidden_state is an attribute defined in the code. However, neither the Decoder class in Page 694 nor the for loop in Page 695 included hidden_state. If I understand correctly, hidden_state may need to be replaced with hidden, which is defined in Page 694 as follows:
![Snipaste_2024-03-27_09-51-15](https://github.com/dvgodoy/PyTorchStepByStep/assets/100295693/efa9763b-275b-4c3a-9bcb-e36ea2f14b66)

As an anverage mind trying to learn PyTorch seriously through this book, may I express some of my personal views on several improvements:
1. Here are two figures from Pages 594-5:
![Snipaste_2024-03-27_10-22-54](https://github.com/dvgodoy/PyTorchStepByStep/assets/100295693/a0597a33-9fa3-4e9d-a5bd-bd52375fe262)
In Figure 8.6, h<sub>0</sub> and h<sub>1</sub> are two different hidden layers. However, In Figure 8.7, h<sub>0</sub> and h<sub>1</sub> are two dimensions of a single hidden layer. If this understanding is correct, I would say it is possbly somewhat misleading to have h<sub>0</sub> and h<sub>1</sub> with different meanings in two consecutive figures. For those acquainted with RNN, this should not be an issue at all. But for beginners of deep learning, I am not sure if I am the only one who would get confused at the first sight (although I was able to correct myself at a second thought).

2. For several examples in Chater 8 and 9, the input features have a dimension of 2 while the dimension of hidden layers is also set to 2. While I was trying to understand the code, I tried to change the hidden layer dimension to some number other than 2. The resulting output gives me an better understanding of the shape of model input and output, which ultimately entails a better understanding of the model. It might be a somewhat suboptimal idea to learn from examples whose feature dimension and hidden dimesion happen to be identical, for green hands.

Please let me know if my suggested typos and suggestions are wrong due to my misunderstanding of the substance knowledge. Thank you.

**Comment by dvgodoy** [comment]
Hi @Patrick-Wen ,

Thank you for pointing those out.

Regarding item number 3, I was referring to the fact of "padding and truncating" because it is the same procedure even if we're using a different max length. I'm not an native speaker either, but the proofreader said it was fine so I kept it like that :-)

Regarding item number 4, I've just run Chapter 11's notebook on Colab and I did not get that error. For some reason, it looks like the following code (from page 894) was not executed on your end:

```python
def is_alice_label(row):
    is_alice = int(row['source'] == 'alice28-1476.txt')
    return {'labels': is_alice}

dataset = dataset.map(is_alice_label)
```
The code creates the missing `labels` key in the dataset and it should make it work as intended. Hope it helps!

Best,
Daniel

**Comment by Patrick-Wen** [comment]
Hi Daniel,

My fault for missing the is_alice_label function. Thank you for the information.

Best regards,
Patrick
